inputTest.py

```diff
@@ -0,0 +1,109 @@
+from register.QuantumRegister import *
+from register.Operators import *
+
+import register.deutsch as deutsch
+import register.deutschJozsa as deutschJozsa
+import register.grovers as grovers
+
+
+def exampleOracle(register):
+
+	def oracle(a):
+		return 0
+	
+	for i in range(register.n_states_):
+		register.state_[i] *= (-1) ** oracle(i)
+
+	return register.state_
+
+
+def objArgTest(Obj,inputs):
+
+	inputType = "FAILED"
+	try:
+		test = Obj(*inputs)
+		#print("SUCCESSFUL INPUT")
+		inputType = "PASSED"
+			
+			
+	except:
+		#	print("FAILED INPUT",type(inputs))
+		test = inputs
+	for i in inputs:
+		print(type(i), end="   ")	
+	print()
+
+	#print(inputType)
+	return inputType
+
+def objSelectTypeTest(Obj,inputs):
+	types = [True,"string",1,2.0]
+	for i in range(len(input)): 
+		try:
+			test = Obj(*inputs)

```

Comment:
why are you unpacking inputs here? are you not checking each input independently?